ui/src/app/components/admin/audit-page-component.tsx

```diff
@@ -91,7 +85,7 @@ const UserInput = ({initialAuditSubject, auditSubjectType, getNextAuditPath, but
   };
 
   return <React.Fragment>
-    {loadNextSubject && <Navigate to={getNextAuditPath(auditSubject)}/>}
+    {navSubject && <Navigate to={getNextAuditPath(auditSubject)}/>}
```

Comment:
Can we  change `<Navigate/>` to take this in as a property like `enabled` so we don't need the &&? I guess I  was thinking the `Navigate` element would be absorbed into a hook as well, but I don't know that I understood.

Reply:
This is the approach I initially took with the hook and had some type issues with it. I wasn't thrilled with the approach and after doing some more reading it seems that having a hook return a component for render is not recommended as it has negative effects on the state management.

I personally do like the `&&` (JS short-circuiting) approach as it is used in the react world quite a bit. Also I feel it is very explicit about state and what will render. In this case navigation occurs when the node is rendered. I feel it makes the code clear when the state is directly tied to whether the component will render - this pattern would apply to all components. Pushing it down into a component and letting the component decide whether it will render requires a bit more investigation when reading the code at a later date.

That said - I could make it look like this (and I do like the look of this/how it reads):
```javascript
   <Navigate when={nextSubjectRequested} to={getNextAuditPath(auditSubject)}/>
```

Overall it would be nice to have a coding style recommendation in place for this. Having `when` be a property we use for conditional rendering of a component could be nice